src/CMakeLists.txt

```diff
@@ -25,6 +25,10 @@ target_compile_definitions(bitcoin_clientversion
   PRIVATE
     HAVE_BUILD_INFO
 )
+target_link_libraries(bitcoin_clientversion
+  PRIVATE
+    core
```

Comment:
We should call this something else, so it's a) more descriptive and b) clear that it is an interface lib. IIRC @theuni had a suggestion, but I don't recall what it was.

Reply:
I'll rename it to `core_interface` in a separated PR.